### PR TITLE
✅ Add e2e test to verify that amp-fit-text recalculates its font size after the box size changes.

### PR DESCRIPTION
**Pull request by rbeckthomas**
This PR adds an end to end test to verify the functionality of the accessibility change made in PR #31317.
This test, changes the box size of an amp-fit-text component and verifies that the font size is recalculated after this change.

### CODE REVIEW

File: extensions/amp-fit-text/0.1/test-e2e/test-amp-fit-text.js [hunk 1]
```diff
@@ -43,10 +43,92 @@ describes.endtoend(
...
+      );
+      const updatedWidth = await controller.getElementCssValue(
+        contentDiv,
+        'width'
+      );
+      await expect(updatedWidth).to.equal('200px');
+
+      await expect(updatedFontSize).to.be.greaterThan(originalFontSize);
+    });
+
```
Comment (alanorozco) [inline extensions/amp-fit-text/0.1/test-e2e/test-amp-fit-text.js:131]: I think this is the same as the first test above (there also shouldn't be an `only()`).

```suggestion
```

**Review by alanorozco — CHANGES_REQUESTED** [review]
A request to simplify.

Should these tests be ported to `1.0` also?

File: extensions/amp-fit-text/0.1/test-e2e/test-amp-fit-text.js [hunk 2]
```diff
@@ -43,10 +43,92 @@ describes.endtoend(
       });
 
       await verifyElementStyles(await selectContentDiv('test4'), {
-        'font-size': '20px',
+        'font-size': '19px',
       });
     });
 
+    describe('user initiated dom changes cause recalculations', function () {
```
Comment (alanorozco) [inline extensions/amp-fit-text/0.1/test-e2e/test-amp-fit-text.js:50]: 1. I think we can use only one amp-fit-text element at `150x150`.
2. Let's rename for purpose.
3. Let's refactor for clarity.

```suggestion
    describe('user initiated dom changes cause recalculations', function () {
      let contentDiv;
      let originalFontSize;

      async function getFontSize(controller, element) {
        return parseInt(
          await controller.getElementCssValue(element, "font-size"),
          10
        );
      }

      async function nextFrame(controller) {
        // If this method doesn't work, use the previous timeout
        await controller.evaluate(
          () =>
            new Promise((resolve) => {
              requestAnimationFrame(resolve);
            })
        );
      }

      beforeEach(async () => {
        contentDiv = await electContentDiv("#recalculate-at-150-150");
        originalFontSize = await getFontSize(controller, contentDiv);
      });

      it("when box size increases font size should increase", async () => {
        await controller.click(
          await controller.findElement("#recalculate-resize-200-200")
        );
        await nextFrame();
        const updatedFontSize = await getFontSize(controller, contentDiv);
        await expect(updatedFontSize).to.be.lessThan(originalFontSize);
      });

      it("when box size decreases font size should decrease", async () => {
        await controller.click(
          await controller.findElement("#recalculate-resize-100-100")
        );
        await nextFrame();
        const updatedFontSize = await getFontSize(controller, contentDiv);
        await expect(updatedFontSize).to.be.lessThan(originalFontSize);
      });
    });
```

With a simpler HTML structure: 

```html
<h3>Recalculate Font Size</h3>
<amp-fit-text
  id="recalculate-at-150-150"
  width="150"
  height="150"
  [width]="recalculateWidth"
  [height]="recalculateHeight"
  class="ft"
>
  Test to verify font size is correctly recalculated.
</amp-fit-text>

<button
  id="recalculate-resize-200-200"
  on="tap:AMP.setState({recalculateWidth: 200, recalculateHeight: 200})"
>
  Resize to 200x200
</button>

<button
  id="recalculate-resize-100-100"
  on="tap:AMP.setState({recalculateWidth: 100, recalculateHeight: 100})"
>
  Resize to 100x100
</button>
```

File: extensions/amp-fit-text/0.1/test/test-amp-fit-text.js [hunk 3]
```diff
@@ -71,6 +72,25 @@ describes.realWin(
...
+      );
+      expect(ft.style.fontSize).to.equal('17px');
+      const setupUpdateFontSizeSpy = env.sandbox.spy(
+        ft.implementation_,
+        'updateFontSize_'
+      );
+      ft.setAttribute('width', '50');
+      ft.setAttribute('height', '100');
+      ft.style.width = '50px';
+      ft.style.height = '100px';
```
Comment (alanorozco) [inline extensions/amp-fit-text/0.1/test/test-amp-fit-text.js:90]: Do we need both of these? Does this work?
```suggestion
      ft.style.width = '50px';
      ft.style.height = '100px';
```

File: extensions/amp-fit-text/0.1/test-e2e/test-amp-fit-text.js [hunk 4]
```diff
@@ -43,10 +43,92 @@ describes.endtoend(
...
+          10
+        );
+        await controller.click(contentButton);
+        // wait for the resizeobserver to recognize the changes
+        // 90ms chosen so that the wait is less than the throttle value for the resizeobserver.
+        await new Promise((resolve) => {
+          setTimeout(() => {
+            resolve();
+          }, 90);
+        });
```
[the remark this answers is not in the document]
Comment (rebeccanthomas) [reply extensions/amp-fit-text/0.1/test-e2e/test-amp-fit-text.js:67]: When I tried using the frame cycle the tests failed, i think the frame cycle resolves sooner than amp-fit-text recalculates the font size.

File: test/fixtures/e2e/amp-fit-text/0.1/amp-fit-text.html [hunk 5]
```diff
@@ -20,9 +20,15 @@
    margin: 8px auto;
    display: block;
   }
+  .textStyle{
+    line-height: 1.5 !important;
+    letter-spacing: 0.12em !important;
+    word-spacing: 0.16em !important;
+  }
```
[the remark this answers is not in the document]
Comment (rebeccanthomas) [reply test/fixtures/e2e/amp-fit-text/0.1/amp-fit-text.html:27]: Ah! yes that was from an earlier version of the test, thank you for pointing that out!

File: extensions/amp-fit-text/0.1/test/test-amp-fit-text.js (hunk 3 above)
Comment (rebeccanthomas) [reply extensions/amp-fit-text/0.1/test/test-amp-fit-text.js:90]: Ah yes! that does work, making the change.

**Review by alanorozco — APPROVED** [review]

**Review by alanorozco — APPROVED** [review]

**Review by rcebulko — APPROVED** [review]
bundle size bump

**Pull request closed without merging**